app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatasetInterface.kt

```diff
@@ -56,4 +58,16 @@ class FirestoreDatasetInterface(
         val pic = query.get().await().toObjects(PictureSchema::class.java).getOrNull(0)
         return pic?.toPublic()
     }
+
+    override suspend fun putPicture(picture: Serializable, category: Category): CategorizedPicture {
```

Comment:
```suggestion
    override suspend fun putPicture(picture: android.net.Uri, category: Category): CategorizedPicture {
```
That's what AddPictureActivity returns and I can work with that.